File: SDKs/amazonaws.com/apprunner/2020-05-15/java/lib/src/main/java/openapisdk/models/shared/EncryptionConfiguration.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * EncryptionConfiguration
 * Describes a custom encryption key that AWS App Runner uses to encrypt copies of the source repository and service logs.
**/
public class EncryptionConfiguration {
    @JsonProperty("KmsKey")
    public String kmsKey;
    public EncryptionConfiguration withKmsKey(String kmsKey) {
        this.kmsKey = kmsKey;
        return this;
    }
}

File: SDKs/amazonaws.com/apprunner/2020-05-15/python/src/sdk/models/shared/encryptionconfiguration.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class EncryptionConfiguration:
    r"""EncryptionConfiguration
    Describes a custom encryption key that AWS App Runner uses to encrypt copies of the source repository and service logs.
    """
    
    kms_key: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('KmsKey') }})
    
